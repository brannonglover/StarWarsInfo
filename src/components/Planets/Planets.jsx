import React, { useState } from 'react'
import { usePaginatedQuery } from "react-query";
import useGetPlanets from "../../hooks/useGetPlanets";
import Planet from '../Planet/Planet';
import Pagination from '../Pagination';
import strings from '../../strings';

const Planets = () => {
  const [page, setPage] = useState(1);
  const { resolvedData, latestData, status } = usePaginatedQuery(["planets", page], useGetPlanets);

  return (
    <>
      <h2>{strings.PLANETS}</h2>
      <Pagination page={page} setPage={setPage} latestData={latestData} />
      {status === "error" && (
        <div>{strings.ERROR}</div>
      )}
      {status === "loading" && (
        <div>Loading...</div>
      )}
      {status === "success" && (
        <>
          {resolvedData.results.map(planet => {
            return <Planet key={planet.name} planet={planet} />
          })}
        </>
      )}
    </>
  )
}

export default Planets;