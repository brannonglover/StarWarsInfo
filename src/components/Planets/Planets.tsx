import React, { useState } from "react";
import { usePaginatedQuery } from "react-query";
import useGetPlanets from "../../hooks/useGetPlanets";
import Planet from "../Planet/Planet";
import Pagination from "../Pagination";
import strings from "../../common/strings";

const Planets = () => {
  const [pageNumber, setPageNumber] = useState(1);
  const { resolvedData, latestData, status } = usePaginatedQuery(
    ["planets", pageNumber],
    useGetPlanets
  );

  return (
    <>
      <h2>{strings.PLANETS}</h2>
      <Pagination
        pageNumber={pageNumber}
        setPageNumber={setPageNumber}
        latestData={latestData}
      />
      {status === "error" && <div>{strings.ERROR}</div>}
      {status === "loading" && <div>Loading...</div>}
      {status === "success" && (
        <>
          {resolvedData.results.map((planet: any) => {
            return <Planet key={planet.name} planet={planet} />;
          })}
        </>
      )}
    </>
  );
};

export default Planets;
