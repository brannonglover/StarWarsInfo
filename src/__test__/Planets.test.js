import React from 'react';
import { cleanup, render } from '@testing-library/react';
import Planets from '../components/Planets/Planets';

afterEach(cleanup);

test('Display Planet component', () => {
  const { asFragment } = render(<Planets />);

  expect(asFragment()).toMatchSnapshot();
});