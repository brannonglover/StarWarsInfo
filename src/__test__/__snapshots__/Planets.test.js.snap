// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display Planet component 1`] = `
<DocumentFragment>
  <h2>
    Planets
  </h2>
  <button
    disabled=""
  >
    Previous
  </button>
  <span>
    1
  </span>
  <button
    disabled=""
  >
    Next
  </button>
  <div>
    Loading...
  </div>
</DocumentFragment>
`;
